front-end/src/app/model/company.model.ts

```diff
@@ -0,0 +1,23 @@
+import { Position } from "./job.model";
+
+export type Company = {
+  id: number;
+  name: string;
+  description: string;
+  founded: Date;
+  location: string;
+  sizeMin: number;
+  sizeMax: number;
+  industryDomainName: string;
+  creationDate: Date;
+  positions: Position[];
+  logoFileName: string;
+};
+
+export type CompanyWithLogo = { logo: string } & Company;
+
+export const isCompanyWithLogo = (
+  company: Company | CompanyWithLogo
+): company is CompanyWithLogo => {
+  return (company as CompanyWithLogo).logo !== undefined;
+};
```

Comment:
This will not work, as `logoFileName ` is a common property between `Company` and `CompanyWithLogo` so this will not  tell us whether the given paremeter is CompanyWithLogo or not. See [type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html) for more information.